go-apps/meep-loc-serv/server/loc-serv.go

```diff
@@ -292,44 +334,74 @@ func Run() (err error) {
 
 // Stop - Stop Location
 func Stop() (err error) {
+	// Stop MEC Service registration ticker
+	if appEnablementEnabled {
+		stopRegistrationTicker()
+	}
+
 	periodicTicker.Stop()
 	return sbi.Stop()
 }
 
-func getAppInstanceId(appName string, appVersion string) string {
-	//var client *appInfoClient.APIClient
-	appInfoClientCfg := appInfoClient.NewConfiguration()
-	appInfoClientCfg.BasePath = appEnablementClientUrl + "/app_info/v1"
+func startRegistrationTicker() {
+	// Wait a few seconds to allow App Enablement Service to start
+	log.Info("Waiting for App Enablement Service to start")
+	time.Sleep(5 * time.Second)
```

Comment:
Don't like this plain sleep.
Prefer the approach where a service tries to access & sleeps if there is a failure.
Sleep should be 1sec or less.

Reply:
Agreed. The problem is that the initial request usually fails with a timeout that is longer (~20 seconds). To avoid pod startup timing issues a sleep was added here. It doesn't eliminate the possible timeout issue (e.g. if the app enablement service pod takes more than 5 seconds to start up), however it usually prevents the initial request from failing & timing out. We could address this by:
1- making sure Location Service only starts after App Enablement service is running (note: currently can't do this with our scenario pods configuration)
2- fail/timeout more quickly on initial request if App Enablement service is not available (i.e. reduce current ~20 second timeout).
3- Keep a sleep in the MEC Services to allow App enablement service to come up (note: does not guarantee App enablement will be running after sleep, in which case we get a ~20 second timeout anyways)

Reply:
After discussion, adding comment in code to clarify the reason for the 5 second sleep.